libkbfs/keybase_service_base.go

```diff
@@ -0,0 +1,454 @@
+	k.userCacheLock.Lock()
+	defer k.userCacheLock.Unlock()
+	delete(k.userCacheUnverifiedKeys, uid)
+}
+
+func (k *KeybaseServiceBase) clearCaches() {
+	k.setCachedCurrentSession(SessionInfo{})
+	k.userCacheLock.Lock()
+	defer k.userCacheLock.Unlock()
+	k.userCache = make(map[keybase1.UID]UserInfo)
+	k.userCacheUnverifiedKeys = make(map[keybase1.UID][]keybase1.PublicKey)
+}
+
+// LoggedIn implements keybase1.NotifySessionInterface.
+func (k *KeybaseServiceBase) LoggedIn(ctx context.Context, name string) error {
+	k.log.CDebugf(ctx, "Current session logged in: %s", name)
+	// Since we don't have the whole session, just clear the cache.
+	k.setCachedCurrentSession(SessionInfo{})
+	if k.config != nil {
+		k.config.MDServer().RefreshAuthToken(ctx)
+		k.config.BlockServer().RefreshAuthToken(ctx)
+		k.config.KBFSOps().RefreshCachedFavorites(ctx)
+	}
+	return nil
+}
+
+// LoggedOut implements keybase1.NotifySessionInterface.
+func (k *KeybaseServiceBase) LoggedOut(ctx context.Context) error {
+	k.log.CDebugf(ctx, "Current session logged out")
+	k.setCachedCurrentSession(SessionInfo{})
+	if k.config != nil {
+		k.config.MDServer().RefreshAuthToken(ctx)
+		k.config.BlockServer().RefreshAuthToken(ctx)
+		k.config.KBFSOps().RefreshCachedFavorites(ctx)
+	}
+	return nil
+}
+
+// KeyfamilyChanged implements keybase1.NotifyKeyfamilyInterface.
+func (k *KeybaseServiceBase) KeyfamilyChanged(ctx context.Context,
+	uid keybase1.UID) error {
+	k.log.CDebugf(ctx, "Key family for user %s changed", uid)
+	k.setCachedUserInfo(uid, UserInfo{})
+	k.clearCachedUnverifiedKeys(uid)
+
+	if k.getCachedCurrentSession().UID == uid {
+		// Ignore any errors for now, we don't want to block this
+		// notification and it's not worth spawning a goroutine for.
+		k.config.MDServer().CheckForRekeys(context.Background())
+	}
+
+	return nil
+}
+
+// PaperKeyCached implements keybase1.NotifyPaperKeyInterface.
+func (k *KeybaseServiceBase) PaperKeyCached(ctx context.Context,
+	arg keybase1.PaperKeyCachedArg) error {
+	k.log.CDebugf(ctx, "Paper key for %s cached", arg.Uid)
+
+	if k.getCachedCurrentSession().UID == arg.Uid {
+		// Ignore any errors for now, we don't want to block this
+		// notification and it's not worth spawning a goroutine for.
+		k.config.MDServer().CheckForRekeys(context.Background())
+	}
+
+	return nil
+}
+
+// ClientOutOfDate implements keybase1.NotifySessionInterface.
+func (k *KeybaseServiceBase) ClientOutOfDate(ctx context.Context,
+	arg keybase1.ClientOutOfDateArg) error {
+	k.log.CDebugf(ctx, "Client out of date: %v", arg)
+	return nil
+}
+
+// ConvertIdentifyError converts a errors during identify into KBFS errors
+func ConvertIdentifyError(assertion string, err error) error {
+	switch err.(type) {
+	case libkb.NotFoundError:
+		return NoSuchUserError{assertion}
+	case libkb.ResolutionError:
+		return NoSuchUserError{assertion}
+	}
+	return err
+}
+
+// Resolve implements the KeybaseService interface for KeybaseDaemonRPC.
```

Comment:
Your comments all still say "KeybaseDaemonRPC", and should probably say "KeybaseServiceBase".